### PR TITLE
(WIP) Implement string format function

### PR DESCRIPTION
**Pull request by ebyhr**
Related to #11343

### CODE REVIEW

File: presto-main/src/test/java/com/facebook/presto/operator/scalar/TestStringFunctions.java [hunk 1]
```diff
@@ -123,6 +123,32 @@ public void testConcat()
...
+        assertFunction("FORMAT(FORMAT('this %s %s', 'is'), 'cool')", VARCHAR, "this is cool");
+
+        // Test varbinary arguments
+        assertFunction("from_utf8(FORMAT(to_utf8('hello %s'), to_utf8('world')))", VARCHAR, "hello world");
+        assertFunction("from_utf8(FORMAT(to_utf8(''), to_utf8('')))", VARCHAR, "");
+
+        // Test concat for non-ASCII
+        assertFunction("FORMAT('hello na\u00EFve %s', 'world')", VARCHAR, "hello na\u00EFve world");
+        assertFunction("FORMAT('\uD801\uDC2D %s', 'end')", VARCHAR, "\uD801\uDC2D end");
+        assertFunction("FORMAT('\uD801\uDC2D%s%s', 'end', '\uD801\uDC2D')", VARCHAR, "\uD801\uDC2Dend\uD801\uDC2D");
```
Comment (findepi) [inline presto-main/src/test/java/com/facebook/presto/operator/scalar/TestStringFunctions.java]: you can use actual letters directly, without Java unicode escapes. https://github.com/prestodb/presto/blob/master/presto-postgresql/src/test/java/com/facebook/presto/plugin/postgresql/TestPostgreSqlTypeMapping.java#L111

**Comment by ebyhr** [comment]
@findepi Thank you for your comment. Updated.

**Comment by wenleix** [comment]
Do you mind re-sending out the PR without the [WIP] in title? (i.e. close this one and create another one)

This is because many community members (including me) ignores PR titled with WIP. And the email sent from GitHub will still show "WIP" even after you removing it. This easily causes PR not get enough attention.

**Comment by ebyhr** [comment]
Sure, I'll do it later. Thank you for teaching me.

**Pull request closed without merging**